kusto/kusto.go

```diff
@@ -130,9 +130,16 @@ func New(endpoint string, auth Authorization, options ...Option) (*Client, error
 		o(client)
 	}
 
-	if err := auth.Validate(endpoint); err != nil {
-		return nil, err
+	if strings.Contains(strings.ToLower(endpoint), ".azuresynapse") {
```

Comment:
WDYT about writing the like this:

```go 
validateEndpoint := endpoint
if strings.Contains(strings.ToLower(endpoint), ".azuresynapse") {
    validateEndpoint = "https://kusto.kusto.windows.net"
}
if err := auth.Validate(validateEndpoint); err != nil {
		return nil, err
}
```

To save on nesting and calling the method twice